### PR TITLE
Retrieve fleet-manager endpoint and cluster ID from environment

### PR DESCRIPTION
**Pull request by mtesseract**
## Description

Currently these values are hard-coded. Make them configurable using environment variables.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing

## Test manual

TODO: Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

**Review by SimonBaeumer — APPROVED** [review]
Nice!

File: fleetshard/config/config.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	"github.com/caarlos0/env/v6"
+	"github.com/pkg/errors"
+)
+
+// Config contains this application's runtime configuration.
+type Config struct {
+	FleetManagerEndpoint string        `env:"FLEET_MANAGER_ENDPOINT" envDefault:"http://127.0.0.1:8000"`
+	ClusterID            string        `env:"CLUSTER_ID"`
+	RuntimePollPeriod    time.Duration `env:"RUNTIME_POLL_PERIOD" envDefault:"1s"`
```
Comment (kovayur) [inline fleetshard/config/config.go:14]: I think 1s period is too small. 60s ?

File: fleetshard/main.go [hunk 2]
```diff
@@ -28,12 +25,19 @@ func main() {
...
 		glog.Info("Unable to set logtostderr to true")
 	}
 
-	runtime, err := runtime.NewRuntime(devEndpoint, clusterID, k8s.CreateClientOrDie())
+	config, err := config.Load()
+	if err != nil {
+		glog.Fatalf("Failed to load configuration: %v", err)
+	}
+
+	glog.Infof("Starting application with configuration: %+v\n", *config)
```
Comment (kovayur) [inline fleetshard/main.go:38]: If in a future we'll add sensitive parameters (secrets), is it ok to print them? 
I mean, such logging may be easily forgotten which may lead to an audit issue.
Is this valid in our case after all?

File: fleetshard/config/config.go (hunk 1 above)
Comment (kovayur) [inline fleetshard/config/config.go:14]: WDYT about `FleetManagerPollPeriod`? I think the intention was to stick to the place where it is used, but out of this context the environment variable name sounds a bit confusing to me.

**Review by kovayur — COMMENTED** [review]
Looks good. Just a few clarifying questions from me.

File: fleetshard/config/config.go (hunk 1 above)
Comment (mtesseract) [reply fleetshard/config/config.go:14]: I don't know, you tell me. :) I just copied what the runtime used to have as a delay until now.

File: fleetshard/main.go (hunk 2 above)
Comment (mtesseract) [reply fleetshard/main.go:38]: You have a point there. I like application printing out the configuration during startup, but, you are right, this should be done in a less ad-hoc manner. Will change and make it a bit more verbose.

File: fleetshard/config/config.go (hunk 1 above)
Comment (mtesseract) [reply fleetshard/config/config.go:14]: This was @SimonBaeumer's idea. I kind of wanted to keep myself out of this naming decision. I can also use your idea.

Comment (kovayur) [reply fleetshard/config/config.go:14]: Thanks, let's keep it for now

File: fleetshard/main.go (hunk 2 above)
Comment (kovayur) [reply fleetshard/main.go]: Thanks!

File: fleetshard/config/config.go (hunk 1 above)
Comment (kovayur) [reply fleetshard/config/config.go:14]: No objections, let's keep as is

**Review by kovayur — APPROVED** [review]
LGTM

**Pull request merged**